src/ComputedFieldTypes.jl

```diff
@@ -161,8 +161,13 @@ function rewrite_new!(e::Expr, tname::Symbol, decl_tvars, def)
                 pfname[1] = make_Type_expr(tname, decl_tvars)
                 curly = e.args[1].args[1]
                 if !(isa(curly, Expr) && curly.head === :curly)
-                    curly = Expr(:curly, curly)
-                    e.args[1].args[1] = curly
+                    curly = Expr(:where, e.args[1])
+                    e.args[1] = curly
+                elseif isa(curly, Expr) && curly.head === :curly
+                    wher = curly.args[2:end]
+                    curly = Expr(:where, Expr(:call, e.args[1].args[1].args[1], e.args[1].args[2]))
+                    e.args[1] = curly
+                    append!(curly.args, wher)
```

Comment:
this change doesn't quite look right. for starters, the comment is now saying the wrong transform—this describes the old structure, while it needs to be altered to reflect the new `where` clauses

Reply:
it should give the correct result based on my testing, but you are right, the naming and labeling is a mixture with the original labeling, so it is more confusing. I will try to clarify it..